### PR TITLE
Do not attempt to load JDK classes from the TCCL

### PR DESCRIPTION
**Pull request by geoand**
This is done because there is no point involving the TCCL to load JDK classes, since these classes can never be loaded by custom classloaders.

I came across this while looking into the generated code of the Reactive REST Client where we can see a lot of:

```java
         ClassLoader var3 = Thread.currentThread().getContextClassLoader();
         Class var4 = Class.forName("java.lang.String", false, var3);
```

which is completely pointless

### CODE REVIEW

File: src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java
Comment (gsmet) [inline src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java:355]: Should it be `! `? Or I misunderstood your intent.

Also I think it was done at some point and @zakkak reverted it: https://github.com/quarkusio/gizmo/pull/126 .

Comment (geoand) [reply src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java]: You are right, the check was wrong.

Comment (geoand) [reply src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java]: And indeed https://github.com/quarkusio/gizmo/pull/126 seems to cause a problem, so let me close this

Comment (geoand) [reply src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java]: Actually, I'll reopen this, but limit ths scope

Comment (geoand) [reply src/main/java/io/quarkus/gizmo/BytecodeCreatorImpl.java]: PR updated

**Review by gsmet — APPROVED** [review]
Looks fine from my side now but let's wait for @stuartwdouglas and @zakkak before merging.

**Review by cescoffier — APPROVED** [review]

**Comment by gsmet** [comment]
@cescoffier hmmm. I said we should wait for @zakkak 's and @stuartwdouglas 's approval :).

**Pull request merged**
